peer-pod-controller/controllers/peerpodconfig_controller.go

```diff
@@ -121,21 +121,35 @@ func (r *PeerPodConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
-// Check if "peer-pods-secret" exists and has valid cloud provider name set in CLOUD_PROVIDER
+// Check if "peer-pods-secret" or "peer-pods-cm" exists and has valid cloud provider name set in CLOUD_PROVIDER
+// logic allows a valid CLOUD_PROVIDER value to be set only in one of the objects, if was set in both, value must be
+// equal and valid
 func (r *PeerPodConfigReconciler) peerpodCloudProviderIsValid() error {
 	peerpodscm := corev1.ConfigMap{}
+	peerpodssecret := corev1.Secret{}
```

Comment:
Do we really need this? What are the issues if we continue to retrieve the provider from the configmap only ?

Reply:
It's not really a problem, but the idea is to avoid the requirement to have both, it's only for simplification purposes